app/scripts/directives/oscWebhookTriggers.js

```diff
@@ -0,0 +1,153 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscWebhookTriggers", function(
+    $uibModal,
+    $filter,
+    APIService) {
+
+    return {
+      restrict: 'E',
+      scope: {
+        webhookSecrets: "=",
+        namespace: "=",
+        type: "@",
+        webhookTriggers: "=",
+        form: '='
+      },
+      templateUrl: 'views/directives/osc-webhook-triggers.html',
+      controller: function($scope) {
+        $scope.secretsVersion = APIService.getPreferredVersion('secrets');
+        $scope.webhookTypesOptions = [{
+          type: 'github',
+          label: 'GitHub'
+        }, {
+          type: 'gitlab',
+          label: 'GitLab'
+        }, {
+          type: 'bitbucket',
+          label: 'Bitbucket'
+        }, {
+          type: 'generic',
+          label: 'Generic'
+        }];
+
+        // Check if the webhook trigger contains only deprecated secret format.
+        $scope.isDeprecated = function(trigger) {
+          var triggerSecretData = $filter('getWebhookSecretData')(trigger);
+          if (_.has(triggerSecretData, 'secret') && !_.has(triggerSecretData, 'secretReference.name')) {
+            return true;
+          }
+          return false;
+        };
+
+        $scope.togglePassword = function(index) {
+          var deprecatedInputField = $('.add-webhook-row')[index].find('.deprecated-secret input');
+          $(deprecatedInputField).attr('type') === 'password' ? $(deprecatedInputField).attr('type', 'text') : $(deprecatedInputField).attr('type', 'password');
+        };
+
+        $scope.removeWebhookTrigger = function(index) {
+          if ($scope.webhookTriggers.length === 1) {
+            var lonelyTrigger = _.first($scope.webhookTriggers);
+            lonelyTrigger.lastTriggerType = "";
+            lonelyTrigger.data = {
+              type: ""
+            };
+          } else {
+            $scope.webhookTriggers.splice(index,1);
+          }
+          $scope.form.$setDirty();
+
+          // After removing trigger check check other triggers for duplicates.
+          _.each($scope.webhookTriggers, function(trigger) {
+            if (trigger.isDuplicit) {
```

Comment:
You always need to check duplicates since the `isDuplicate` will be false the first time it appears.

```
trigger A
trigger B (duplicate of A)
```

We need to unflag trigger B if trigger A is removed.

Reply:
Thats what Im doing in the `checkDuplicates` (renamed `duplicityCheck`) function which I feed with trigger I want to check, the one that is removed or the one user modifies 
Basically what Im doing there filtering an array of `matchingTriggers`  where all are basically duplicates, and I set the first ones `isDuplicate` field to `false` and on all others I set the field to `true`

Reply:
I see. I missed that you are looping over all of them and not just checking the is duplicate flag on the one you're removing in this spot.

Reply:
yes, already updated the logic and will only loop on the trigger the action is on